File: fib.ex
defmodule Fib do 
  def fib(0) do 0 end
  def fib(1) do 1 end
  def fib(n) do fib(n-1) + fib(n-2) end
end

start = Time.utc_now
IO.puts Fib.fib(42)
IO.puts Time.diff(Time.utc_now,  start)

File: fib.rs
use std::time::{Instant};

fn main() {
  let start = Instant::now();
  println!("{}",fib(42));
  let duration = start.elapsed();
  println!("Time elapsed is: {:?}", duration);
}

fn fib(n: i32) -> i32 {
    if n == 0 { return 0; }
    if n == 1 { return 1; }

    fib(n - 1) + fib(n - 2)
}
